domain-http/interface/src/main/java/org/jboss/as/domain/http/server/ManagementHttpServer.java

```diff
@@ -245,7 +236,10 @@ private static void setupOpenListener(HttpOpenListener listener, ModelController
         }
 
         ManagementRootConsoleRedirectHandler rootConsoleRedirectHandler = new ManagementRootConsoleRedirectHandler(consoleHandler);
-        DomainApiCheckHandler domainApiHandler = new DomainApiCheckHandler(modelController, controlledProcessStateService, allowedOrigins);
+        HttpHandler domainApiHandler = InExecutorHandler.wrap(
+                managementExecutor,
+                new DomainApiCheckHandler(modelController, controlledProcessStateService, allowedOrigins)
+        );
```

Comment:
This is a nit; ignore it unless you need to touch this PR for some other reason:

This block would be better starting at L249, right before the HttpHandler readinessHandler... line. That groups the rootConsoleRedirectHandler logic together and then the code starts on the domainApiHandler.